### PR TITLE
BUG/ENH: Add single field filtering

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by llllllllll** [comment]
Is this just an optimization or does this not currently work?

**Comment by cpcloud** [comment]
The first test case `t.amount[t.amount > 0]` doesn't work

**Comment by cpcloud** [comment]
@llllllllll Any more comments? Would like to merge this soon

File: blaze/compute/sql.py [hunk 1]
```diff
@@ -176,6 +176,12 @@ def compute_up(t, s, **kwargs):
     return getattr(t, 'op', getattr(safuncs, sym, getattr(sa.func, sym)))(s)
 
 
+@dispatch(Selection, sa.sql.ColumnElement)
+def compute_up(expr, data, scope=None, **kwargs):
+    predicate = compute(expr.predicate, data, post_compute=False)
+    return sa.select([data]).where(predicate)
```
Comment (llllllllll) [inline blaze/compute/sql.py]: should we use the `select` function from this module here?

**Comment by llllllllll** [comment]
one question but lgtm

File: blaze/compute/sql.py (hunk 1 above)
Comment (cpcloud) [inline blaze/compute/sql.py]: i'm -1 on using that function as it hides a subtle aspect of sqlalchemy that can sometimes cause weird issues if you're not very careful about nesting selects

Comment (cpcloud) [inline blaze/compute/sql.py]: I think in all cases I know the kind of thing I'm getting back (`Select`, `Table`, `ColumnElement`, etc)

Comment (cpcloud) [inline blaze/compute/sql.py]: I'll probably remove that function in an upcoming PR

Comment (llllllllll) [inline blaze/compute/sql.py]: Sounds good. I just wanted to make sure that this wasn't solving some subtle issue that has been discovered in the past. I would also be in favor of removing at it is confusing when used with `sa.select`

**Pull request merged**